Sources/HummingbirdCore/Server/HTTP/HTTPChannelHandler.swift

```diff
@@ -33,83 +33,77 @@ enum HTTPChannelError: Error {
     case closeConnection
 }
 
-enum HTTPState: Int, AtomicValue {
-    case idle
-    case processing
-    case cancelled
-}
-
 extension HTTPChannelHandler {
     public func handleHTTP(asyncChannel: NIOAsyncChannel<HTTPRequestPart, HTTPResponsePart>, logger: Logger) async {
-        let processingRequest = ManagedAtomic(HTTPState.idle)
         do {
-            try await withGracefulShutdownHandler {
-                try await asyncChannel.executeThenClose { inbound, outbound in
-                    let responseWriter = HBHTTPServerBodyWriter(outbound: outbound)
-                    var iterator = inbound.makeAsyncIterator()
-
-                    // read first part, verify it is a head
-                    guard let part = try await iterator.next() else { return }
-                    guard case .head(var head) = part else {
-                        throw HTTPChannelError.unexpectedHTTPPart(part)
-                    }
-
-                    while true {
-                        // set to processing unless it is cancelled then exit
-                        guard processingRequest.exchange(.processing, ordering: .relaxed) == .idle else { break }
```

Comment:
I didn't add the graceful check here, I wasn't sure if that was "correct" because we already received `.head`